### DISCUSSION TITLE
Send a dynamic data which is not in the form

### DISCUSSION THREAD

**Discussion by Fidia-e** [emailjs-com/help#11] [Q&A]
Hi there, 

I can send a static value in a hidden input, but I'd like to know if it's possible to do the same thing with a value witch is updated on form submission  ?

I'm working with ReactJs.
I've tried with useState() and useRef() but the value keeps the initial state or nothing.

Thanks for help

**Comment by xr0master** [comment]
I think you are doing everything right, if you are using the form and you need to add some data for the submitted data, then hidden input fields are the best solution.

Or you can create a state (by useState), save all changes there (onChange event), and then send this state through the "emailjs.send" method.

**Comment by Fidia-e** [comment]
Okay, I will try and let you know, thank you !

**Comment by Fidia-e** [comment]
Okay, I think the problem is that I'm using React-Hook-Form 
And the data, I want to catch and send via EmailJS, isn't update in a input

**Comment by Fidia-e** [comment]
Hi,

I fixed it this way : 
- kept my hidden input
- set initial value in `defaultValues` of the `useFrom()` function of React-hook-Form
- use `setValue()` to set the value I wanted to send
- use `trigger()` of React-hook-Form too
- and finally `{...register("myInputValue")}`

And voilà
No useState needed

Thanks again for help
